src/mesh/mesh_tools.C

```diff
@@ -1027,8 +1031,37 @@ void MeshTools::clear_spline_nodes(MeshBase & mesh)
 
 
 
-#ifdef DEBUG
-void MeshTools::libmesh_assert_equal_n_systems (const MeshBase & mesh)
+bool valid_is_prepared (const MeshBase & mesh)
+{
+  LOG_SCOPE("libmesh_assert_valid_is_prepared()", "MeshTools");
+
+  if (!mesh.is_prepared())
+    return true;
+
+  std::unique_ptr<MeshBase> mesh_clone = mesh.clone();
+
+  // Try preparing (without allowing repartitioning or renumbering, to
+  // avoid false assertion failures)
+  bool old_allow_renumbering = mesh_clone->allow_renumbering();
+  mesh_clone->allow_renumbering(false);
+  bool old_allow_remote_element_removal =
+    mesh_clone->allow_remote_element_removal();
+  bool old_skip_partitioning = mesh_clone->skip_partitioning();
+  mesh_clone->skip_partitioning(true);
+  mesh_clone->allow_remote_element_removal(false);
+  mesh_clone->prepare_for_use();
+  mesh_clone->allow_renumbering(old_allow_renumbering);
+  mesh_clone->allow_remote_element_removal(old_allow_remote_element_removal);
+  mesh_clone->skip_partitioning(old_skip_partitioning);
```

Comment:
```suggestion
```

Reply:
Definitely leaving these in; if the settings are changed and left that way then operator== afterward returns false.